org.opent2t.sample.thermostat.superpopular/com.smartthings.thermostat/js/thingTranslator.js

```diff
@@ -137,20 +139,20 @@ function providerSchemaToPlatformSchema(providerSchema, expand) {
 
     var hvacMode = createResource('oic.r.mode', 'oic.if.a', 'hvacMode', expand, {
         supportedModes: ['coolOnly', 'heatOnly', 'auto', 'off'],
-        modes: [deviceHvacModeToTranslatorHvacMode(providerSchema['attributes'].thermostatMode)]
+        modes: [deviceHvacModeToTranslatorHvacMode(attributes.thermostatMode)]
     });
 
     var hasFan = createResource('oic.r.sensor', 'oic.if.s', 'hasFan', expand, {
-        value: providerSchema['attributes'].thermostatFanMode !== undefined && providerSchema['attributes'].thermostatFanMode !== null
+        value: attributes.thermostatFanMode !== undefined && attributes.thermostatFanMode !== null
```

Comment:
I think this also does the same thing and is simpler:
value: ((typeof attributes.thermostatFanMode) !== "undefined")

Reply:
@rgruian - typeof null is object. So you need to check for null and undefined separately

Reply:
value: attributes.thermostatFanMode ? true : false